app/Http/Controllers/Auth/AuthController.php

```diff
@@ -87,19 +87,36 @@ public function handleProviderCallback($provider)
                 // stroing data to our use table and logging them in
                 $username = $user->getEmail();
                 $first_name = $user->getName();
+                $last_name = $user->getLastname();
+                $role = $user->getRole();
+
                 if ($user->nickname) {
                     $username = $user->nickname;
                 }
                 if (!$first_name) {
                     $first_name = $username;
                 }
-                $data = [
-                    'first_name' => $first_name,
-                    'email'      => $user->getEmail(),
-                    'user_name'  => $username,
-                    'role'       => 'user',
-                    'active'     => 1,
-                ];
+                if ($role == 'admin' || $role == 'agent') {
+                    $data = [
+                        'first_name' => $first_name,
+                        'last_name'  => $last_name,
+                        'email'      => $user->getEmail(),
+                        'user_name'  => $username,
+                        'role'       => $role,
+                        'active'     => 1,
+                        'primary_dpt'     => 1,
+                        'assign_group'     => 1,
+                    ];    
+                } else {
+                    $data = [
+                        'first_name' => $first_name,
+                        'last_name'  => $last_name,
+                        'email'      => $user->getEmail(),
+                        'user_name'  => $username,
+                        'role'       => $role,
+                        'active'     => 1,
+                    ];    
+                }
```

Comment:
I think it could be better to leave the $data creation as it was, adding/changing missing properties afterwards, doing like 
```php
$data['last_name'] = $last_name;
``` 
(if a key doesn't exist in a PHP map it's automatically added)